js-api-doc/compile.d.ts

```diff
@@ -162,3 +252,67 @@ export function compileStringAsync(
   source: string,
   options?: StringOptions<'async'>
 ): Promise<CompileResult>;
+
+/**
+ * Creates an instance of a synchronous {@link Compiler}. Each compiler instance
+ * exposes the {@link compile} and {@link compileString} methods within the
+ * lifespan of the Compiler. Given identical input, these methods will return
+ * results identical to their counterparts exposed at the module root. To use
+ * asynchronous compilation, use {@link initAsyncCompiler}.
+ *
+ * When using the [sass-embedded] npm package, this allows Sass to reuse a
+ * process across multiple compilations, reducing the amount of time needed to
+ * create and destroy each process.
```

Comment:
The second half of this sentence isn't totally accurate—you're not creating and destroying "each process" at all with a `Compiler` object, there's only one process the entire time. It might be better to just say something like "This is much faster than calling the top-level compile methods multiple times when using the sass-embedded npm package". The most important thing is the speed, the reason why is kind of an implementation detail.

Reply:
Because there are multiple variables (number of compilations, implementation, compiler/top level), I went with the following in https://github.com/sass/sass/pull/3744/commits/5d4329bd5ff43007697a400245835e3951704a65 -

```
 * When calling the compile functions multiple times, using a compiler instance
 * with the [sass-embedded] npm package is much faster than using the top-level
 * compilation methods or the [sass] npm package.
 ```